tests/conftest.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Pytest configuration."""
+
+from __future__ import absolute_import, print_function
+
+import os
+import shutil
+import sys
+import tempfile
+from contextlib import contextmanager
+
+import pytest
+from flask import Flask
+from mock import patch
+from pkg_resources import EntryPoint
+
+
+@pytest.fixture()
+def app():
+    """Flask application fixture."""
+    app = Flask('testapp')
+    app.config.update(
+        TESTING=True
+    )
+    return app
+
+
+def create_file_hierarchy(root_dir_path, files):
+    """Create a hierarchy of files in a directory.
+
+    :param root_dir_path: directory in which all files will be created
+    :param files: dict of <relative file path> -> <file content>
+    """
+    for path, content in files.items():
```

Comment:
`shutils.copytree ()` copies a directory into another, which means that the source directory must exist in the first place. This function factorizes the creation of directory from a `dict` so that it can be used by `dir_factory` and `pkg_factory`. Having a dict enables generation on the fly and easy testing as we don't have to read the files in the tests.